Lesson 5

borisshavrin opened the pull request.

borisshavrin committed:
Revert "Competed lesson-5"

This reverts commit a299466231d5fafeed79ea1d2507639eec475863.

borisshavrin committed:
Competed lesson-5

borisshavrin committed:
Completed lesson-5

borisshavrin committed:
Update task1.py

borisshavrin commented on task7.py:
```diff
@@ -0,0 +1,35 @@
+        new_line = line.strip().split()                     # разделяем строку на отдельные элементы
+        name, form, cash, cost = new_line[0], new_line[1], int(new_line[2]), int(new_line[3])   # отдельные эл-ты
+        profit = cash - cost                                # вычисляем прибыль для фирмы
+        firm_stat = [name, form, cash, cost, profit]        # сохраняем элементы и значение прибыли в список
```
Список firm_stat было необязательно вводить и  без него код более читабельным становится:
if profit > 0:
    total_profit += profit
    count += 1
d1[name] = profit

pantonov-gb commented on task1.py:
```diff
@@ -0,0 +1,11 @@
+    while True:
+        if x == '':
+            break                                   # остановка цикла при вводе пустой строки
+        else:
```
Необязательный else, можно убрать, код станет получше

pantonov-gb commented on task3.py:
```diff
@@ -0,0 +1,18 @@
+        res = line.strip().split()      # очищаем строку от лишних символов, сохраняем ее в отдельном списке
+        data.append(res)                # добавляем каждую строку в data
+
+    total_cash = 0                      # суммарный оклад всех сотрудников
```
После того, как файл прочитан и больше нам не нужен - можно выходить из контекстного менеджера

pantonov-gb commented on task7.py:
```diff
@@ -0,0 +1,35 @@
+        name, form, cash, cost = new_line[0], new_line[1], int(new_line[2]), int(new_line[3])   # отдельные эл-ты
+        profit = cash - cost                                # вычисляем прибыль для фирмы
+        firm_stat = [name, form, cash, cost, profit]        # сохраняем элементы и значение прибыли в список
+        if firm_stat[4] > 0:                    # пополняем счетчик неотрицательной прибыли и фирм с неотр. прибылью
```
Сравните с `if profit > 0:`. Какой код понятнее?

pantonov-gb reviewed (approved):
Ок

borisshavrin committed:
Update task7.py

Удален список firm_stat - без него код более читабельным становится:
if profit > 0:
    total_profit += profit
    count += 1
d1[name] = profit
